src/components/Comments/ReplyCommentItem.js

```diff
@@ -158,16 +162,28 @@ const ReplyCommentItem = ({
       onLongPress();
     }
   };
-
   const username = comment?.data?.anon_user_info_color_name
-    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    ? `${comment.data.anon_user_info_color_name} ${comment.data?.anon_user_info_emoji_name}`
     : user?.data?.username;
 
+  const onPressDm = async () => {
+    try {
+      refSheet.current.open();
+      setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+      const data = await getAllowAnonDmStatus('comment', comment?.id);
+      setUserAllowDm(data?.user.allow_anon_dm);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+    }
+  };
+
   const onPressDM = async () => {
     try {
       setLoading({...loading, loadingDm: true});
       if (!comment?.data?.anon_user_info_color_name) {
-        const channelName = [username, profile?.myProfile?.username].join(',');
+        const channelName = username;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `username` logic has been updated to include the anonymous user's color name along with the emoji name. The `onPressDm` function has been introduced which fetches the status of whether the user allows anonymous direct messages or not. The `onPressDM` function has been updated to include the `feedId` in the `sendMessageDM` function call. All these changes are necessary for the new feature being implemented.
```diff
+ const username = comment?.data?.anon_user_info_color_name
+ ? `${comment.data.anon_user_info_color_name} ${comment.data?.anon_user_info_emoji_name}`
+ : user?.data?.username;
+ const onPressDm = async () => {
+   try {
+     refSheet.current.open();
+     setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+     const data = await getAllowAnonDmStatus('comment', comment?.id);
+     setUserAllowDm(data?.user.allow_anon_dm);
+   } catch (e) {
+     console.log(e);
+   } finally {
+     setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+   }
+ };
+ const onPressDM = async () => {
+   try {
+     setLoading({...loading, loadingDm: true});
+     if (!comment?.data?.anon_user_info_color_name) {
+       const channelName = username;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->